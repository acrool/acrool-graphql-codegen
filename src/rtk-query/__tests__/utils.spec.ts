import {parse, buildSchema} from 'graphql';
import {plugin} from '../index';



const basicDoc = parse(/* GraphQL */ `
    query testQuery {
        feed {
            id
            commentCount
            repository {
                full_name
                html_url
                owner {
                    avatar_url
                }
            }
        }
    }
`);

describe('My Plugin', () => {


    const schema = buildSchema(/* GraphQL */ `
        type Query {
            workspacesWithPagination: String!
            workspaces: String!
            workspace: String!
        }

        type Mutation {
            workspaceCreate: WorkspaceCreateInput!
        }
        type PaginateInput {
            name: String
        }
        type WorkspaceCreateInput {
            ownerId: String
        }

    `);
    const docs = [
        {
            document: basicDoc,
        },
        // {
        //     document: basicMutation,
        // },
    ];

    const usedBefore = process.memoryUsage().heapUsed;

    it('Should greet', async () => {
        const result = await plugin(schema, docs, {
            // name: 'bear-react-query'
            importBaseApiFrom: '@/library/graphql/baseApi',
            importBaseApiAlternateName: 'baseApi',
            exportHooks: true,
            exportApi: true,
            noExportDocument: false,
        });

        const usedAfter = process.memoryUsage().heapUsed;
        console.log(`Memory used by the function: ${(usedAfter - usedBefore) / 1024 / 1024} MB`);

        expect(result).toMatchSnapshot();
    });
});
