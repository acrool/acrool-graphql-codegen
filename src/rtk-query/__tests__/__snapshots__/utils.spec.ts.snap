// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My Plugin Should greet 1`] = `
{
  "content": "
export const TestQueryDocument = \`
    query testQuery {
  feed {
    id
    commentCount
    repository {
      full_name
      html_url
      owner {
        avatar_url
      }
    }
  }
}
    \`;",
  "prepend": [
    "import { baseApi } from '@/library/graphql/baseApi';",
    "import {gql, useSubscription, SubscriptionHookOptions} from '@apollo/client';",
  ],
}
`;
